File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        sort(strs.begin(),strs.end());
        int n=strs.size()-1,i=0;
        string ans="";
        while(i<strs[0].length()&&i<strs[n].length())
        {
            if(strs[0][i]==strs[n][i])
            ans+=strs[0][i];
            else
            break;
            i++;
        }
        return ans;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        String a=strs[0],b=strs[strs.length-1],ans="";
        System.out.println(a);
        System.out.println(b);
        int i=0,j=0;

        while(i<a.length()&&j<b.length()){
            if(a.charAt(i)==b.charAt(j))
            ans+=a.charAt(i);
            else
            return ans;
            i++;
            j++;
           
        }
        return ans;

    }
}